components/service/experiments/src/main/java/mozilla/components/service/experiments/util/VersionString.kt

```diff
@@ -0,0 +1,73 @@
+package mozilla.components.service.experiments.util
+
+import java.lang.IllegalArgumentException
+import kotlin.math.max
+
+internal class VersionString(
```

Comment:
Isn't there a lot of overlap to [`EngineVersion`](https://github.com/mozilla-mobile/android-components/blob/master/components/concept/engine/src/main/java/mozilla/components/concept/engine/utils/EngineVersion.kt) here?
We shouldn't duplicate logic if we can avoid it. Why not share this as single utility code somewhere?

Reply:
I'll file a follow-up bug and discuss it with the A-C team

Reply:
I was worried about creating a cross-dependency right now

Reply:
Filed the bug to follow-up [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1562241).  I don't think that this blocks or hurts us to proceed with this functionality living in service-experiments for now.